app/src/cards/DisparityBarChartCard.tsx

```diff
@@ -10,45 +10,65 @@ import { Fips } from "../utils/madlib/Fips";
 import {
   BreakdownVar,
   BREAKDOWN_VAR_DISPLAY_NAMES,
-  MetricToggle,
-  shareOf,
-  METRIC_FULL_NAMES,
-  METRIC_SHORT_NAMES,
 } from "../utils/madlib/DisplayNames";
 import useDatasetStore from "../data/useDatasetStore";
 import { Breakdowns } from "../data/Breakdowns";
 import { getDependentDatasets, VariableId } from "../data/variableProviders";
 import VariableQuery from "../data/VariableQuery";
+import { MetricConfig, VariableConfig } from "../data/MetricConfig";
 
 import CardWrapper from "./CardWrapper";
 
-export type ChartToggle = "percents" | "ratio";
+const VALID_METRIC_TYPES = ["pct_share", "per100k"];
+
+function getInitalMetricConfig(variableConfig: VariableConfig) {
+  return variableConfig.metrics["pct_share"]
+    ? variableConfig.metrics["pct_share"]
+    : variableConfig.metrics["per100k"];
+}
 
 function DisparityBarChartCard(props: {
   breakdownVar: BreakdownVar;
-  metricId: MetricToggle;
+  variableConfig: VariableConfig;
+  nonstandardizedRace: boolean /* TODO- ideally wouldn't go here, could be calculated based on dataset */;
   fips: Fips;
 }) {
-  const [chartToggle, setChartToggle] = useState<ChartToggle>("percents");
+  const [metricConfig, setMetricConfig] = useState<MetricConfig>(
+    getInitalMetricConfig(props.variableConfig)
+  );
+  useEffect(() => {
+    setMetricConfig(getInitalMetricConfig(props.variableConfig));
+  }, [props.variableConfig]);
```

Comment:
I think this may cause issues because useEffect can run in a subsequent react update. For example, you could have a VariableConfig that doesn't match with the selected metric config which could cause UI bugs or even crash the page.

Several options I can think of to fix:
- push metricConfig up to the parent class and update it by passing in a setter in
- make useState a type union of variableConfig and metricConfig, and have useEffect set both together

I think the latter technically still could have issues like if breakdownVar differs between the old/new variables, so I think pushing the metricConfig to the parent is maybe better

Reply:
added a TODO for this afternoon